src/app/qgsmaptooloffsetcurve.cpp

```diff
@@ -71,15 +72,25 @@ void QgsMapToolOffsetCurve::canvasReleaseEvent( QgsMapMouseEvent *e )
     deleteRubberBandAndGeometry();
     mGeometryModified = false;
 
-    QgsPointLocator::Match match = mCanvas->snappingUtils()->snapToCurrentLayer( e->pos(),
-                                   QgsPointLocator::Types( QgsPointLocator::Edge | QgsPointLocator::Area ) );
+    QgsPointLocator::Match match;
 
-    if ( ( match.hasEdge() || match.hasArea() ) && match.layer() )
+    if ( e->modifiers() & Qt::ControlModifier )
     {
-      mLayer = match.layer();
+      match = mCanvas->snappingUtils()->snapToMap( e->pos(), nullptr );
```

Comment:
This line won't work because of PR #31374

See my comment : https://github.com/qgis/QGIS/pull/31648#issuecomment-529652349